variable/subspan_view.cpp

```diff
@@ -4,113 +4,53 @@
 /// @author Simon Heybrock
 #include "scipp/variable/subspan_view.h"
 #include "scipp/core/except.h"
+#include "scipp/variable/cumulative.h"
+#include "scipp/variable/shape.h"
 #include "scipp/variable/transform.h"
+#include "scipp/variable/util.h"
 
 namespace scipp::variable {
 
 namespace {
 
-/// Helper returning vector of subspans from begin to end
-template <class Var, class T>
-auto make_subspans(Var &, const ElementArrayView<T> &first,
-                   const ElementArrayView<T> &last) {
-  const auto len = first.size();
-  std::vector<
-      span<std::conditional_t<std::is_const_v<Var>, std::add_const_t<T>, T>>>
-      spans;
-  spans.reserve(len);
-  for (scipp::index i = 0; i < len; ++i)
-    spans.emplace_back(scipp::span(&first[i], &last[i] + 1));
-  return spans;
-}
-
-template <class T>
-auto make_empty_subspans(const ElementArrayView<T> &, const Dimensions &dims) {
-  return std::vector<span<T>>(dims.volume());
-}
-
 template <class T>
 auto make_subspans(T *base, const Variable &indices,
                    const scipp::index stride) {
-  const auto spans = variable::transform<scipp::index_pair>(
+  if (stride != 1)
+    throw std::logic_error(
+        "span only supports stride=1, this should be "
+        "unreachable due to an earlier check, may want to generalize this "
+        "later to support in particular stride=0 for broadcasted buffers");
```

Comment:
A bit of a strange error message.

Reply:
It is not user facing, just as an extra safety check.